GSoC26_H/hitl/hitl_app.py

```
@@ -0,0 +1,402 @@
+
+def sync_to_github(new_decisions):
+    """Push new decisions to GitHub via the Contents API. Requires
+    github_token in Streamlit secrets. Returns (success, message)."""
+    token = st.secrets.get("github_token")
+    if not token:
+        return False, "No github_token found in app secrets. Add it under Settings → Secrets."
+
+    api_url = f"https://api.github.com/repos/{GITHUB_REPO}/contents/{GITHUB_FILE_PATH}"
+    headers = {"Authorization": f"token {token}", "Accept": "application/vnd.github+json"}
+
+    resp = requests.get(api_url, headers=headers, params={"ref": GITHUB_BRANCH})
```

Ruff (0.15.21)

[error] Probable use of `requests` call without timeout

(S113)